tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs

```diff
@@ -77,52 +76,75 @@ public static RecordEntry Deserialize(JsonElement element)
+        private static void DeserializeBody(RequestOrResponse requestOrResponse, in JsonElement property)
         {
             if (property.ValueKind == JsonValueKind.Null)
             {
-                return null;
+                requestOrResponse.Body = null;
             }
 
-            if (IsTextContentType(headers, out Encoding encoding))
+            if (IsTextContentType(requestOrResponse.Headers, out Encoding encoding))
             {
-                if (property.ValueKind == JsonValueKind.Object)
-                {
-                    using var memoryStream = new MemoryStream();
-                    using var writer = new Utf8JsonWriter(memoryStream, writerOptions);
-                    property.WriteTo(writer);
-                    writer.Flush();
-                    return memoryStream.ToArray();
-                }
-                else if (property.ValueKind == JsonValueKind.Array)
-                {
-                    StringBuilder stringBuilder = new StringBuilder();
+                byte[] bytes = null;
 
-                    foreach (JsonElement item in property.EnumerateArray())
+                if (property.ValueKind == JsonValueKind.Array)
+                {
                     {
-                        stringBuilder.Append(item.GetString());
-                    }
+                        StringBuilder sb = new StringBuilder();
+                        foreach (JsonElement item in property.EnumerateArray())
+                        {
+                            sb.Append(item.GetString());
+                        }
 
-                    return encoding.GetBytes(stringBuilder.ToString());
+                        bytes = encoding.GetBytes(sb.ToString());
+                    }
+                }
+                else if (property.ValueKind == JsonValueKind.String)
+                {
+                    bytes = encoding.GetBytes(property.GetString());
                 }
                 else
                 {
-                    return encoding.GetBytes(property.GetString());
+                    bytes = encoding.GetBytes(property.GetRawText());
                 }
+
+                // TODO consider versioning RecordSession so that we can stop doing the below for newly created recordings
+                NormalizeJsonBody(requestOrResponse, bytes);
+            }
+            else if (property.ValueKind == JsonValueKind.Array)
+            {
+                requestOrResponse.Body = Array.Empty<byte>();
             }
 
-            if (property.ValueKind == JsonValueKind.Array)
+            else
             {
-                return Array.Empty<byte>();
+                requestOrResponse.Body = Convert.FromBase64String(property.GetString());
             }
+        }
 
-            return Convert.FromBase64String(property.GetString());
+        public static void NormalizeJsonBody(RequestOrResponse requestOrResponse, byte[] bytes)
```

Comment:
Why do you pass bytes in?

Reply:
Can we use only the RequestOrResponse?

Reply:
when deserializing the body will not be populated on the requestOrResponse

Reply:
So what about populating it right before calling the method?

Reply:
I suppose I can just set it.